StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs

```diff
@@ -44,7 +50,195 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            context.RegisterSyntaxNodeAction(HandleEventDeclaration, SyntaxKind.EventDeclaration);
+            context.RegisterSyntaxNodeAction(HandleEventFieldDeclaration, SyntaxKind.EventFieldDeclaration);
+        }
+
+        private void HandleTypeDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            BaseTypeDeclarationSyntax declaration = context.Node as BaseTypeDeclarationSyntax;
+
+            bool isDefinedInClassOrStruct = IsDefinedInClassOrStructRecursive(declaration.Parent);
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, isDefinedInClassOrStruct ? SyntaxKind.PrivateKeyword : SyntaxKind.PublicKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandleMethodDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            MethodDeclarationSyntax declaration = context.Node as MethodDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandleConstructorDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            ConstructorDeclarationSyntax declaration = context.Node as ConstructorDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            PropertyDeclarationSyntax declaration = context.Node as PropertyDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
```

Comment:
The default visibility of a property declaration within an interface is `public`.